source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,471 @@
+by specifying which traffic is allowed to *and* from the resources - e.g. VMs -
+using high level abstraction L3 and L4 semantics.
+
+Furthermore, security groups allow these rules to be applied in fine
+grained fashion - e.g. the user can specify which rules apply to which
+VMs.
+
+Below, you can find some examples of what is the feature's ambition:
+* **only** allow incoming traffic to a specific VM from a specific
+CIDR
+* **only** allow incoming access to a range of destination ports in a
+set of VMs
+* **only** allow outgoing traffic to a specific port on a set of VMs
+
+Having said that, and highlighting that L2 protection is **outside of
+scope** of security groups, it could be further enhanced by implementing the
+[Allowed address pairs](https://developer.openstack.org/api-ref/network/v2/#allowed-address-pairs)
+Networking API extension.
+
+## Objectives
+
+* Provide a security group implementation, using OVN as the backend.
+No additional backend / database should be used other than ovn's
+northbound database.
+
+* When possible, Openstack neutron's behavior should be mimicked: by
+default, all VM incoming IP *and* outgoing traffic is dropped. This behavior
+is better specified in the [requirements](#requirements) section.
+
+* The security group feature should be configurable, and activated by
+default - this means an option for using oVirt's external networks
+without security group support should be possible.
+
+* If the external network has a subnet defined, DHCP traffic is allowed to
+/ from the VM.
+
+### Requirements
+
+* For ingress traffic (to an instance)
+  + Only traffic matched with security group rules are allowed.
+  + When there is no rule defined, all traffic are dropped.
+* For egress traffic (from an instance)
+  + Only traffic matched with security group rules are allowed.
+  + When there is no rule defined, all egress traffic are dropped.
+  + When a new security group is created, rules to allow all egress traffic are automatically added.
+
+### Security groups description
+
+Security groups, in the general sense, are nothing more than containers
+for [security group rules](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules). 
+They are applied to logical ports, and thus are used to associate a set of 
+rules to a port.
+
+They are are indeed a white list - e.g. everything is *denied* by
+default, and the user adds rules that allow traffic. As long as *one*
+rule is matched, the traffic is allowed.
+
+Security group rules can match on L3 and L4 parameters, specifically on
+the protocol (L3), the IP protocol (L4), and destination ports. The remote IP
+address is also a subject for the match conditions, and rules can be
+added to allow *incoming* and *outgoing* traffic.
+
+The API of security group rules can be found [here](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules).
+
+### Activating the feature
```

Comment:
Do we need additional configuration for this?
Can the "port_security_enabled" attribute of the port not be used for this? With this we can decide at network/port level.

Reply:
+1
We should also consider the Neutron behaviour in this.
Meaning before and after you apply the first SG.